src/mock.rs

```diff
@@ -0,0 +1,648 @@
+
+}
+
+impl super::Trait for NftRegistryTest {
+    type Event = MetaEvent;
+    // type Balance = u64;
+    type Randomness = RandomnessCollectiveFlip;
+    type NFTDepositBase = NFTDepositBase;
+    type NFTDepositPerByte = NFTDepositPerByte;
+    type NFTValidationRegistryDeposit = NFTValidationRegistryDeposit;
+    type Currency = Balances;
+}
+
+pub struct DummyContractAddressFor;
+impl ContractAddressFor<H256, u64> for DummyContractAddressFor {
+    fn contract_address_for(_code_hash: &H256, _data: &[u8], origin: &u64) -> u64 {
+        *origin + 1
+    }
+}
+
+pub struct DummyTrieIdGenerator;
+impl TrieIdGenerator<u64> for DummyTrieIdGenerator {
+    fn trie_id(account_id: &u64) -> TrieId {
+        use sp_core::storage::well_known_keys;
+
+        let new_seed = contracts::AccountCounter::mutate(|v| {
+            *v = v.wrapping_add(1);
+            *v
+        });
+
+        let mut res = vec![];
+        res.extend_from_slice(well_known_keys::CHILD_STORAGE_KEY_PREFIX);
+        res.extend_from_slice(b"default:");
+        res.extend_from_slice(&new_seed.to_le_bytes());
+        res.extend_from_slice(&account_id.to_le_bytes());
+        res
+    }
+}
+
+pub struct DummyComputeDispatchFee;
+impl ComputeDispatchFee<Call, u64> for DummyComputeDispatchFee {
+    fn compute_dispatch_fee(call: &Call) -> u64 {
+        69
+    }
+}
+
+pub const ALICE: u64 = 1;
+pub const BOB: u64 = 2;
+pub const CHARLIE: u64 = 3;
+pub const DJANGO: u64 = 4;
+pub const NULL_CONTRACT: u64 = 100;
+pub const INVALID_UID: u64 = 100;
+
+pub fn print_all_events() {
+    println!("------------------- Print Events Started -------------------");
+    for event in <system::Module<NftRegistryTest>>::events() {
+        println!("{:?}", event);
+    }
+    println!("------------------- Print Events Ended -------------------");
+}
+// data part, 02 is module index. three modules defined
+// balances::Balances,
+// contracts::Contract,
+// nftregistry::NftRegistry,
+// 02 is function index. to check the index in NFT call enum.
+pub const CODE_DISPATCH_CALL: &str = r#"
+(module
+    (import "env" "ext_dispatch_call" (func $ext_dispatch_call (param i32 i32)))
+    (import "env" "memory" (memory 1 1))
+    (func (export "call")
+        (call $ext_dispatch_call
+            (i32.const 8) ;; Pointer to the start of encoded call buffer
+            (i32.const 10) ;; Length of the buffer
+        )
+    )
+    (func (export "deploy"))
+    ;; Generated by codec::Encode::encode(&Call::NftRegistry(nftregistry::Call::finish_mint(registry_id))))
+    (data (i32.const 8) "\02\02\00\00\00\00\00\00\00\00")
+)
+"#;
+
+pub fn compile_module<T>(
+    wabt_module: &str,
+) -> std::result::Result<(Vec<u8>, <T::Hashing as Hash>::Output), wabt::Error>
+where
+    T: system::Trait,
+{
+    let wasm = wabt::wat2wasm(wabt_module)?;
+    let code_hash = T::Hashing::hash(&wasm);
+    Ok((wasm, code_hash))
+}
+
+pub fn create_account_mock(account_id: u64) {
```

Comment:
Whats the reason for all those functions having `mock` in their name? The actions are actually executed, not mocked, right? I think it would be clearer to call them `_test`.

Reply:
I put mock in function name for two reasons. 1. it is defined in mock file. 2. The functions executed, but they are executed in a mocked externality environment.
_test is fine, it is also distinct.

Reply:
Thanks!